src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -103,9 +105,12 @@ LogicalMessage[] Extract(IncomingMessage physicalMessage)
                 }
             }
 
+         
             var messageTypes = messageMetadata.Select(metadata => metadata.MessageType).ToList();
             var messageSerializer = deserializerResolver.Resolve(physicalMessage.Headers);
 
+            mapper.EnsureTypesAreInitialized(messageTypes);
```

Comment:
Also the serializerz missuse the mapper for their own purposed for non message types so we want to make sure that only message types is added. Eg. the json serializer relies on the mapper returning null for non message types.